meteoblue_dataset_sdk/client.py

```diff
@@ -162,23 +165,36 @@ async def queryRaw(self, params: dict):
 
     async def query(self, params: dict):
         """
-        Query meteoblue dataset api asynchronously, transfer data using protobuf and return a structured object
-        :param params: query parameters, see https://docs.meteoblue.com/en/apis/environmental-data/dataset-api
+        Query meteoblue dataset api asynchronously, transfer data using protobuf and
+        return a structured object
+
+        :param params:
+            query parameters,
+            see https://docs.meteoblue.com/en/apis/environmental-data/dataset-api
         :return: DatasetApiProtobuf object
         """
 
         params["format"] = "protobuf"
-
+        if self.cache:
+            cached_query_results = await self.cache.get(params)
+            if cached_query_results:
+                msg = DatasetApiProtobuf()
+                msg.ParseFromString(cached_query_results)
+                return msg
         async with self.queryRaw(params) as response:
             data = await response.read()
+            if self.cache:
+                await self.cache.set(params, data)
             msg = DatasetApiProtobuf()
             msg.ParseFromString(data)
             return msg
 
     def querySync(self, params: dict):
```

Comment:
By the way: I would seriously change the casing of these methods and variables, its anti pattern and pretty confusing for anyone else.
lower case and underscore for function and method, camel case for classes. 
https://www.python.org/dev/peps/pep-0008/#function-and-variable-names

Reply:
Yes, you can rename it, BUT keep also the old function name which call the new one. Otherwise it will be a breaking change